src/content/1.7/modules/creation/module-translation/classic-system.md

```diff
@@ -0,0 +1,377 @@
+{if isset($my_module_name) && $my_module_name}
+   {$my_module_name}
+{else}
+   {l s='World' mod='mymodule'}
+{/if}
+!
+```
+
+Isn't there a better way to make this work?
+
+We have two possible cases:
+
+1. `"Hello " . $my_module_name . "!"`
+2. `"Hello World!"`
+
+We can start improving this by keeping the first case, then replacing the variable content to "World" as needed:
+
+```
+{if !isset($my_module_name) || !$my_module_name}
+  {capture name='my_module_tempvar'}{l s='World' mod='mymodule'}{/capture}
+  {assign var='my_module_name' value=$smarty.capture.my_module_tempvar}
+{/if}
+{l s='Hello,' mod='mymodule'} {$my_module_name$}!
+```
+
+This is a start! In this example, when `$my_module_name` is not defined, we capture the value of translating "World" and assign that value in `$my_module_name`. This way, we can simply display `$my_module_name` in any case.
+
+Now, when working in internationalization, it's better not to make assumptions regarding spacing and positioning of variables. For example, in French, there should be a non-breaking space before the exclamation mark (`!`), but not in other languages. How can we handle this?
+
+In PrestaShop, you can interpolate variables within your wordings using using [sprintf replacement markers](https://php.net/sprintf), such as `%s` or `%1$s`. Using this feature, we could get rid of the explicit concatenation of "Hello, ", "$my_module_name" and "!", and let it depend on the translated wording:
+
+```
+{if !isset($my_module_name) || !$my_module_name}
+  {capture name='my_module_tempvar'}{l s='World' mod='mymodule'}{/capture}
+  {assign var='my_module_name' value=$smarty.capture.my_module_tempvar}
+{/if}
+{l s='Hello %s!' sprintf=[$my_module_name] mod='mymodule'}
+```
+
+This way, since the variable replacement will happen _after_ the wording is translated, we can have a wording in English like `Hello %s!`, being translated as `Bonjour %s !` in French.
+
+Also, you can use sprintf markers (like `%s` and `%d`) or replacement tokens (like `[1]` and `%something%`). If you chose the latter, just provide an associative array: 
+`sprintf=['[1]' => 'some replacement, '%something%' => 'something else']`
+
+#### Interpolating HTML
+
+You may need to add HTML content in your translated string. Writing it directly in the string (original or translated) won't work, as the special characters would be escaped to avoid XSS security issues.
+
+Instead, you should use placeholders and replace them with HTML code.
+
+Let's take an example with a link in a string, which can be tricky to do. 
+The first solution that comes to mind would be to concatenate the translated strings with raw HTML code.
+But as we explained before, this solution is not recommended, because the words order could be different depending on the translation language.
+
+```php
+{l
+  s='If you want a category to appear in the menu of your shop, go to [1]Modules > Modules & Services > Installed modules.[/1] Then, configure your menu module.'
```

Comment:
couldn't we introduce a new best practice here? WDYT about that:

```
[modules-link]Modules > Modules & Services > Installed modules.[/modules-link]
```

I'm not convinced it will work as I hope.

Reply:
Yes it should work 👍